# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/quota/StorageManager.cpp
# Commit: a8c5441fd896
# Full Hash: a8c5441fd89658dea45d939b1e930cf3d148fafd
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-01-26 21:47:24
# Description:
#   Bug 1873573 - Rename PromiseWorkerProxy::WorkerPromise() to GetWorkerPromise(). r=dom-worker-reviewers,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199323
# ==============================================================================

diff -r 533ecc686bac -r a8c5441fd896 dom/quota/StorageManager.cpp
--- a/dom/quota/StorageManager.cpp	Fri Jan 26 10:41:43 2024 +0000
+++ b/dom/quota/StorageManager.cpp	Fri Jan 26 10:48:56 2024 +0000
@@ -427,20 +427,15 @@
     promise = mPromise;
   } else {
     MOZ_ASSERT(mProxy);
-
-    // The worker ref might have been notified already before we are run.
-    MutexAutoLock lock(mProxy->Lock());
-    if (!mProxy->CleanedUp()) {
-      promise = mProxy->WorkerPromise();
-
-      // Only clean up for worker case.
-      autoCleanup.emplace(mProxy);
+    promise = mProxy->GetWorkerPromise();
+    if (!promise) {
+      return;
     }
+    // Only clean up for worker case.
+    autoCleanup.emplace(mProxy);
   }
 
-  if (!promise) {
-    return;
-  }
+  MOZ_ASSERT(promise);
 
   if (mType == Type::Estimate) {
     if (NS_SUCCEEDED(mResultCode)) {
